### ISSUE TITLE
fsck saving metadata mysql error

### ISSUE THREAD

**Issue by HD-CMS** [gboudreau/Greyhole#170]
Hello again ;-)

with the updated version now greyhole works correct. So I started now an fsck to recreate the symlinks (new file and folder structure on OS install). The scan completed and all jobs are created:

```
             Write   Delete   Rename    Check
Arbeit           0        0        0  122,298
XXXXX          0        0        0    1,170
Bilder           0        0        0   33,571
XXXXX         0        0        0  170,226
Dokumente        0        0        0   17,596
Musik            0        0        0   37,338
Public           0        0        0   40,006
XXXXX          0        0        0      656
Videos           0        0        0    9,726
=============================================
Total            0        0        0  432,587
             Write   Delete   Rename    Check

The following is the number of pending operations that the Greyhole daemon still needs to parse.
Until it does, the nature of those operations is unknown.
Spooled operations that have been parsed will be listed above and disappear from the count below.

Spooled    1,417
```

Now I had another problem when the jobs being executed. See this log after try to restart:

```
Feb 18 15:58:35 INFO daemon: Greyhole (version 0.10.1) daemon started.
Feb 18 15:58:35 INFO daemon: Optimizing MySQL tables...
Feb 18 15:58:35 DEBUG daemon: Loading metadata store backup directories...
Feb 18 15:58:35 DEBUG daemon:   Found 2 directories in the settings table.
Feb 18 15:58:35 DEBUG daemon: Checking symlink at /usr/lib/x86_64-linux-gnu/samba/vfs/greyhole.so...
Feb 18 15:58:35 DEBUG daemon:   Is OK.
Feb 18 15:58:35 DEBUG read_smb_spool: Processing Samba spool...
Feb 18 15:58:35 ERROR read_smb_spool: PHP Fatal Error: Uncaught Exception: SQLSTATE[01000]: Warning: 1265 Data truncated for column 'complete' at row 1 in /usr/bin/greyhole:619
Stack trace:
#0 /usr/bin/greyhole(5117): DB::execute('UPDATE tasks SE...', Array)
#1 /usr/bin/greyhole(5020): close_task('fwrite', 'Videos', '39', Array)
#2 /usr/bin/greyhole(3599): parse_samba_spool()
#3 /usr/bin/greyhole(3574): DaemonRunner->initialize()
#4 /usr/bin/greyhole(3608): DaemonRunner->run()
#5 {main}
  thrown; BT: greyhole[L619] 
```

I would say that this has to do with the new mysql structure. 

Thanks
Dennis

**Comment by gboudreau** [comment]
Somehow related; latest code just skips all DB migrations that was always running before. And one of those migration was to modify the complete column, and that change is missing from the schema file. Will fix.

**Comment by HD-CMS** [comment]
Hmm strange... after two another restarts (my plan was to print the complete mysql $query from line 5117) the recreation of the symlinks goes forward. I will watch. Currently 417,409 files left.

**Comment by HD-CMS** [comment]
Okay, found the error. It´s not an error from the fsck, but from regular access over samba. If I had looked right, that would have been clear. (Processing Samba spool...)

**Comment by gboudreau** [comment]
Thanks for the report.
Fixed in 0.10.2

**Comment by HD-CMS** [comment]
Thanks for your great support!
